File: src/pixel.ts
import { Color } from "./color"

export class Pixel {
	private color: Color
	private index: number
	
	constructor(color: Color, index: number) {
		this.color = color;
		this.index = index;
	}

	public getColor(): Color {
		return this.color;
	}

	public getIndex(): number {
		return this.index;
	}

	public toJson(): Object {
		return {
			index: this.index,
			data: this.color.toJson()
		}
	}
	public getString(): string {
		return JSON.stringify(this.toJson());
	}
}

File: CSharpApp/Pixel.cs


namespace CSharpApp
{

	class Pixel
	{
		public Color color { get; }
		public int index { get; }

		public Pixel(Color color, int index)
		{
			this.color = color;
			this.index = index;
		}

		public override string ToString()
		{

			return $"Pixel - Index: {this.index} --- {this.color.ToString()}";

		}

	}


}